modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/OnboardingUpgradeFeaturesPage.kt

```diff
@@ -111,69 +138,400 @@ internal fun OnboardingUpgradeFeaturesPage(
+                ) { titleRes ->
+                    TextH20(
+                        text = stringResource(titleRes),
+                        textAlign = TextAlign.Center,
+                        color = Color.White,
+                        modifier = Modifier
+                            .padding(horizontal = 24.dp)
+                            .fillMaxWidth(),
+                    )
+                }
 
-                IconRow(Modifier.padding(horizontal = 24.dp))
+                Spacer(Modifier.height(24.dp))
 
-                Spacer(Modifier.height(36.dp))
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(bottom = 24.dp),
+                    contentAlignment = Alignment.Center
+                ) {
+                    StyledToggle(
+                        state.subscriptionFrequencies
+                            .map { stringResource(id = it.localisedLabelRes) },
+                    ) {
+                        onSubscriptionFrequencyChanged(it)
+                    }
+                }
 
-                TextH10(
-                    text = stringResource(LR.string.onboarding_upgrade_everything_you_love_about_pocket_casts_plus),
-                    color = Color.White,
-                    modifier = Modifier.padding(horizontal = 24.dp),
+                FeatureCards(
+                    state = state,
+                    onFeatureCardChanged = onFeatureCardChanged,
+                    onUpgradePressed = onUpgradePressed,
+                    canUpgrade = canUpgrade,
+                    upgradePrice = upgradePrice,
                 )
+            }
 
-                Spacer(Modifier.height(12.dp))
+            Spacer(Modifier.weight(1f))
+        }
+    }
+}
 
-                TextP30(
-                    text = stringResource(LR.string.onboarding_upgrade_exclusive_features_and_options),
-                    color = Color.White.copy(alpha = 0.8f),
-                    modifier = Modifier.padding(horizontal = 24.dp),
-                )
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun FeatureCards(
+    state: OnboardingUpgradeFeaturesState,
+    onFeatureCardChanged: (Int) -> Unit,
+    onUpgradePressed: () -> Unit,
+    canUpgrade: Boolean,
+    upgradePrice: (String) -> String,
+) {
+    val pagerState = rememberPagerState()
 
-                Spacer(Modifier.height(58.dp))
+    LaunchedEffect(pagerState) {
+        snapshotFlow { pagerState.currentPage }.collect { index ->
+            onFeatureCardChanged(index)
+        }
+    }
 
-                FeatureRow(scrollAutomatically = state.scrollAutomatically)
+    HorizontalPager(
+        pageCount = state.featureCards.size,
+        state = pagerState,
+        pageSize = PageSize.Fixed(LocalConfiguration.current.screenWidthDp.dp - 64.dp),
+        contentPadding = PaddingValues(horizontal = 32.dp),
+    ) { index ->
+        FeatureCard(
```

Comment:
Ah, `IntrisicSize.Max` probably doesn't work because the pager is lazy. I forgot about that. 🤔 I'm definitely fine with trying to address this separately later.